llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4372,6 +4372,9 @@ void LoopVectorizationPlanner::emitInvalidCostRemarks(
   SmallVector<RecipeVFPair> InvalidCosts;
   for (const auto &Plan : VPlans) {
     for (ElementCount VF : Plan->vectorFactors()) {
+      if (VF.isScalar())
```

Comment:
Could you add a comment here why we skip scalar VFs? Presumably because we don’t expect them to have invalid costs?

Reply:
Updated comments.

Currently, we cannot query the vplan-based cost model with scalar VF because we use lots of `cast<VectorType>(toVectorTy(SrcTy, VF))` in `computeCost()`.